_rules/lists-follow-content-model-a73be2.md

```diff
@@ -0,0 +1,277 @@
+---
+id: a73be2
+name: List elements follow content model
+rule_type: atomic
+description: |
+  This rule checks that list elements have child nodes allowed by the content model
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility tree
+  - CSS Styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Wilco Fiers
+  previous_authors:
+    - Jey Nandakumar
+---
+
+## Applicability
+
+This rule applies to any [HTML elements][] with a [hidden state][] of "false", where the [semantic role][] is the same as the [implicit role][], and for which one of the following is true:
+
+- **lists**: The element is a `ul`, `ol` or `menu`; or
+- **definition lists**: The element is a `dl`, or a `div` that is a [child][] in the [flat tree][] of a `dl`.
+
+## Expectation
+
+For each [child][] in the [flat tree][] of each test target, one of the following is true:
+
+- **implicit role**: the [child][] follows the [content model][] and has no [explicit role][]; or
+- **explicit role**: the [child][] has an [explicit role][] which is the [implicit ARIA semantics][] for an HTML element included in the [content model][] for the test target; or
+- **hidden**: the [child][] is not [included in the accessibility tree][].
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+All children of `ul`, `ol`, and `menu` elements must be either an element with a [semantic role][] of `listitem`, or a node that is hidden, such as a `script` or `template` element, an HTML comment, or a text node that consists of only [ascii whitespace][].
+
+All children of a `dl` element must be an element with a [semantic role][] of `term` or `definition`, a `div` with no [explicit role][], or a node that is hidden. If the child has a `definition` role, it must have a previous sibling that has a `term` role. If the child is a `div` element with no [explicit role][] it is treated as a `term` and `definition` group. The children of the `div` have the same requirement as the `dl` element, except that it does not allow another `div`.
```

Comment:
There is a bit of confusion between the content model (which is pure HTML) and the roles (which HTML ignores). This is normal given what we want the rule to do, and I agree that the rule should do that. The fact that the rule name and description is only about content model (which I also agree is good) doesn't clarify this.

I think we can start the Background section with the explanation of this difference. Something like
> Content model is pure HTML, this rule accept ARIA replication of the content model through roles. Cases that pass **explicit role** may actually fail the HTML content model. They pass the rule because the overall structure (e.g. a list with items) is good.

Plus, maybe we need an assumption that in `<ul><div>Foo</div></ul>` the `div` is styled to appear as an item in the list, otherwise there is no structure conveyed through presentation and 1.3.1 doesn't apply.
(this also mean that Failed Examples doing this need a bit of styling (`ul > div::before { content: "·" }` should be enough))

---
Anyway, the point is that these notes keep are helpful to understand what the rule checks but don't really clear the confusion. When saying that "children of `ul` must have a role of `listitem`", we talk about the rule, not the content model (which doesn't care about role). I think we should make it explicit that this "must" is about the rule.

Reply:
Fair point. I've written something. Let me know what you think.